types/Support/Collection.php

```diff
@@ -1049,13 +1049,20 @@ function ($collection, $count) {
 assertType('Illuminate\Support\Collection<int, User>', $collection->splice(1));
 assertType('Illuminate\Support\Collection<int, User>', $collection->splice(1, 1, [new User]));
 
-assertType('Illuminate\Support\Collection<int, User>', $collection->transform(function ($user, $int) {
+assertType('mixed', $collection->transform(function ($user, $int) {
```

Comment:
This change should have been an indicator that something was wrong---phpstan should be perfectly capable of determining the correct generics